Source/WebCore/rendering/RenderVideo.cpp

```diff
@@ -191,10 +193,11 @@ IntRect RenderVideo::videoBox() const
     if (videoElement().shouldDisplayPosterImage())
         intrinsicSize = m_cachedImageSize;
 
-    if (videoElement().isFullscreen() && areAspectRatiosEssentiallyEqual(intrinsicSize, contentSize(), page().deviceScaleFactor()))
+    auto videoBoxRect = snappedIntRect(replacedContentRect(intrinsicSize));
+    if (inElementOrVideoFullscreen() && contentSizeAlmostEqualsFrameSize(view().frameView().layoutSize(), videoBoxRect.size(), page().deviceScaleFactor()))
```

Comment:
Two changes from 266284@main are the above line, `contentSize()` replaced by `videoBoxRect.size()`